src/test/java/org/myrobotlab/service/ServiceScriptTest.java

```diff
@@ -62,45 +56,57 @@ public void tearDown() throws Exception {
         ServiceType st = sts.get(i);
         serviceTypes.add(st.getSimpleName());
         System.out.println(System.getProperty("user.dir") + "/src/main/resources/resource/" + st.getSimpleName() + "/" + st.getSimpleName() + ".py");
-        String script = FileIO.toString("src/main/resources/resource/" + st.getSimpleName() + "/" + st.getSimpleName() + ".py"); // GitHub.getPyRobotLabScript(branch, st.getSimpleName());
+        String script = FileIO.toSafeString("src/main/resources/resource/" + st.getSimpleName() + "/" + st.getSimpleName() + ".py"); // GitHub.getPyRobotLabScript(branch,
+                                                                                                                                     // st.getSimpleName());
         if (script != null) {
           pythonScripts.put(st.getSimpleName(), script);
         } else {
           log.warn("{} needs a service script", st.getSimpleName());
           servicesNeedingPythonScripts.add(st.getSimpleName());
         }
       }
-      Set<String> gitHubServiceScripts = GitHub.getServiceScriptNames();
-      for (String key : gitHubServiceScripts) {
-        String serviceName = key.substring(0, key.lastIndexOf("."));
-        if (!serviceTypes.contains(serviceName)) {
-          log.info("script exists {} but service does not", key);
-          pythonScriptsWithNoServiceType.add(key);
-        }
-      }
+
       log.info("remove scripts - script found but no service type {}", pythonScriptsWithNoServiceType);
     }
     return pythonScripts;
   }
 
   @Test
-  public void test() throws Exception {
-    String[] b = new String[] { "LeapMotion", "OpenNi", "Runtime", "SlamBad", "_TemplateService", "Cli", "WebGui", "JMonkeyEngine", "ImageDisplay", "GoogleAssistant",
+  public void testServiceScripts() throws Exception {
+    String[] b = new String[] { "Agent", "LeapMotion", "OpenNi", "Runtime", "SlamBad", "_TemplateService", "Cli", "WebGui", "JMonkeyEngine", "ImageDisplay", "GoogleAssistant",
         "PickToLight", "PythonProxy", "Sprinkler", "_TemplateProxy", "SwingGui" };
     Set<String> blacklist = new HashSet<String>(Arrays.asList(b));
 
-    pythonScripts = getPyRobotLabServiceScripts(branch);
+    pythonScripts = getPyRobotLabServiceScripts();
     log.info("testing {} scripts", pythonScripts.size());
 
     for (String name : pythonScripts.keySet()) {
+      
+      name = "AudioFile";
```

Comment:
Um... this is wrong...  why are we always overriding and forcing "AudioFile" to be tested here.

Reply:
heh ... 
Question should be .. "Why are we always ignoring or disabling service scripts"  
Heh .. ya that's right all service script testing has been ignored or disabled for a very long time .. I'm in the process of activating it all back up ... but I can't fix all of the bad scripts at once ..
This is diagnostics ... and you can be assured that the build will fail when I pull out the sole service I have been testing, and let it test all it should be - but I can't fix it all at once ...
I'll pull it out and let builds fail until we need a build ..

FIXED - builds will now fail and build nodes will now kill themselves until action is taken ... need more elves !